client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java

```diff
@@ -965,11 +976,11 @@ public final void fieldCapsAsync(FieldCapabilitiesRequest fieldCapabilitiesReque
      * layer has been added to the ReST client, and requests should extend {@link Validatable} instead of {@link ActionRequest}.
      */
     @Deprecated
-    protected final <Req extends ActionRequest, Resp> Resp performRequestAndParseEntity(Req request,
-                                                                            CheckedFunction<Req, Request, IOException> requestConverter,
-                                                                            RequestOptions options,
-                                                                            CheckedFunction<XContentParser, Resp, IOException> entityParser,
-                                                                            Set<Integer> ignores) throws IOException {
+    public final <Req extends ActionRequest, Resp> Resp performRequestAndParseEntity(Req request,
```

Comment:
One of the side-effects of these methods being public is that they will be suggested to our users by IDEs although they will never be what users need to call, also they become part of our public API while we considered them internal until now. I wish there was a way to not make them public. Maybe we should consider keeping the clients all on the same package? I think that having the other clients subclass this class is a no-go given that the main client needs to have knowledge of all of the clients.